src/primitives/transaction.cpp

```diff
@@ -161,3 +161,29 @@ size_t CTransaction::GetTxSize() const
         nTxSize = ::GetSerializeSize(*this, SER_NETWORK, CTransaction::CURRENT_VERSION);
     return nTxSize;
 }
+
+
+bool CTransaction::HasData()
+{
+    for (auto out : vout)
+    {
```

Comment:
use a reference here ?
-for (auto out : vout)
+for (auto &out : vout)

same below...

Reply:
thx good catch!